test/results_test.dart

```diff
@@ -741,4 +741,42 @@ Future<void> main([List<String>? args]) async {
     final items = realm.query<Product>(ids);
     expect(items.length, 0);
   });
+
+  test('RealmResults.first throws on empty', () async {
+    final config = Configuration.local([Dog.schema, Person.schema]);
+    final realm = getRealm(config);
+
+    final empty = Iterable<int>.empty();
+    expect(() => empty.first, throwsA(isA<StateError>()));
```

Comment:
why do we test the state's Inerrable behavior? I suggest we test only the Realm implementation

Reply:
Just to illustrate behaviour for regular `Iterable`.. will add a comment